fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalJoin.java

```diff
@@ -250,4 +250,9 @@ public LEFT_CHILD_TYPE left() {
     public RIGHT_CHILD_TYPE right() {
         return (RIGHT_CHILD_TYPE) child(1);
     }
+
+    public LogicalJoin withHashJoinConjuncts(List<Expression> hashJoinConjuncts) {
```

Comment:
I think you can add a method named with `withHashJoinConjunctsAndChidren`, so that you could finish the creation of new JoinNode in one call.